web_invalid_tab/static/src/js/view_form.js

```diff
@@ -0,0 +1,62 @@
+openerp.web_invalid_tab = function(instance) {
+
+    var tab_selector = 'div[role="tabpanel"]';
+
+    function tab_link(tab) {
+        return $("a[href='#" + tab.attr('id') + "']");
+    }
+
+    function is_visible(tab, element) {
+        if ($(element).hasClass('oe_form_invisible') || element.style.display == 'none') {
+            return false;
+        }
+        else if (element.parentNode && element.parentNode != tab) {
+            return is_visible(tab, element.parentNode);
+        }
+        else {
+            return true;
+        }
+    }
+
+    instance.web.form.AbstractField.include({
+        _check_css_flags: function() {
+            if (this.field.translate) {
+                this.$el.find('.oe_field_translate').toggle(this.field_manager.get('actual_mode') !== "create");
+            }
+            if (!this.disable_utility_classes) {
+                if (this.field_manager.get('display_invalid_fields')) {
+                    this.$el.toggleClass('oe_form_invalid', !this.is_valid());
+                    this._check_invalid_tab();
+                }
+            }
+        },
+        _check_invalid_tab: function() {
```

Comment:
if the name is _*check*_invalid_tab then we are expecting an idempotent function but we can find {remove|add}Class.